presto-cli/src/main/java/com/facebook/presto/cli/Query.java

```diff
@@ -315,6 +316,16 @@ public void renderFailure(PrintStream out)
         out.println();
     }
 
+    public void renderWarnings(PrintStream out)
+    {
+        QueryStatusInfo results = client.finalStatusInfo();
+        results.getWarnings().stream()
+                .forEach(warning -> out.printf("Query %s warning %s: %s", results.getId(), warning.getWarningCode(), warning.getMessage()));
```

Comment:
The query id is unnecessary here -- it will be the same for all lines. This already takes place within the context of a query execution.